lib/Emitter.php

```diff
@@ -61,16 +62,14 @@ public function emit($value): Promise {
             }
 
             /**
-             * Resolves the stream with the given value.
-             *
-             * @param mixed $value
+             * Completes the iterator.
              */
-            public function resolve($value = null) {
-                ($this->resolve)($value);
+            public function complete() {
```

Comment:
I like `complete()`, less risk of method name collision with other things that _might_ also implement `Iterator`.